app/client/templates/elements/compileContract.js

```diff
@@ -38,27 +47,66 @@ Template['elements_compileContract'].onCreated(function() {
                     template.aceEditor.focus();
             });
         }
+
     });
 
     // update and generate the contract data 
     this.autorun(function() {
+        
+        // selected contract
         var selectedContract = TemplateVar.get('selectedContract');
         var constructorInputs = _.clone(TemplateVar.get('constructorInputs'));
+        var selectedToken = TemplateVar.getFrom('.select-token', 'selectedToken');
+        var mainRecipient = TemplateVar.getFrom('div.dapp-address-input input.to', 'value');
+        var replayProtectionOn = TemplateVar.get('replay-protection-checkbox');
+        var selectedType = TemplateVar.get('selectedType');
+
+        if(selectedType == 'source-code' && selectedContract){        
```

Comment:
this should be a tripple ===, which means same content and type.
Its just good practice and would throw an error in ads lint